### ISSUE TITLE
PEP 3109 Use new syntax for raising exceptions

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8345]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)| @rodrigc reported|
|-|-|
|Trac ID|trac#8345|
|Type|enhancement|
|Created|2016-05-22 08:40:58Z|
|Branch|https://github.com/twisted/twisted/tree/8345-rodrigc-pep3109|

PEP 3109 ( https://www.python.org/dev/peps/pep-3109/ ) specifies a new syntax how to catch exceptions which works in Python 2.6 and higher.

The old syntax:
```
raise E, V
```

should be changed to:
```
raise E(V)
```

Both syntaxes work in Python 2.6 and Python 2.7
Only the new syntax works in Python 3.x

<details><summary>Searchable metadata</summary>

```
trac-id__8345 8345
type__enhancement enhancement
reporter__rodrigc rodrigc
priority__normal normal
milestone__Python_3_x Python-3.x
branch__8345_rodrigc_pep3109 8345-rodrigc-pep3109
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__None None
time__1463906458136556 1463906458136556
changetime__1464178287300075 1464178287300075
version__None None
owner__rodrigc rodrigc

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_1"></a>|@rodrigc commented|
|-|-|

See:
https://github.com/twisted/twisted/pull/66

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_3"></a>|@adiroiban set owner to @rodrigc|
|-|-|

changes look good. thanks!

I have pushed the news fragment.

will wait for test to pass and will merge.

thanks again for the cleanup

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_4"></a>|@adiroiban commented|
|-|-|

test are failing and this branch can not be merged until all builders are green.

I have sent the branch again to the tests.

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_5"></a>|@adiroiban commented|
|-|-|

here are the failures

```
FAIL]
Traceback (most recent call last):
  File "/buildslave/debian8-py2.7/Twisted/build/py27-tests/local/lib/python2.7/site-packages/twisted/trial/_asynctest.py", line 75, in _eb
    raise self.failureException(output)
twisted.trial.unittest.FailTest: 
Expected: (<type 'exceptions.ValueError'>,)
Got:
[Failure instance: Traceback: <type 'exceptions.AttributeError'>: 'exceptions.ValueError' object has no attribute 'with_traceback'
/buildslave/debian8-py2.7/Twisted/build/py27-tests/local/lib/python2.7/site-packages/twisted/internet/utils.py:197:runWithWarningsSuppressed
/buildslave/debian8-py2.7/Twisted/build/py27-tests/local/lib/python2.7/site-packages/twisted/test/test_adbapi.py:802:test_runWithConnectionRaiseOriginalError
/buildslave/debian8-py2.7/Twisted/build/py27-tests/local/lib/python2.7/site-packages/twisted/enterprise/adbapi.py:292:runWithConnection
/buildslave/debian8-py2.7/Twisted/build/py27-tests/local/lib/python2.7/site-packages/twisted/internet/threads.py:53:deferToThreadPool
--- <exception caught here> ---
/buildslave/debian8-py2.7/Twisted/build/py27-tests/local/lib/python2.7/site-packages/twisted/test/test_adbapi.py:725:callInThreadWithCallback
/buildslave/debian8-py2.7/Twisted/build/py27-tests/local/lib/python2.7/site-packages/twisted/enterprise/adbapi.py:307:_runWithConnection
]

twisted.test.test_adbapi.ConnectionPoolTests.test_runWithConnectionRaiseOriginalError
===============================================================================
[FAIL]
Traceback (most recent call last):
  File "/buildslave/debian8-py2.7/Twisted/build/py27-tests/local/lib/python2.7/site-packages/twisted/trial/_asynctest.py", line 75, in _eb
    raise self.failureException(output)
twisted.trial.unittest.FailTest: 
Expected: (<type 'exceptions.ValueError'>,)
Got:
[Failure instance: Traceback: <type 'exceptions.AttributeError'>: 'exceptions.ValueError' object has no attribute 'with_traceback'
/buildslave/debian8-py2.7/Twisted/build/py27-tests/local/lib/python2.7/site-packages/twisted/internet/utils.py:197:runWithWarningsSuppressed
/buildslave/debian8-py2.7/Twisted/build/py27-tests/local/lib/python2.7/site-packages/twisted/test/test_adbapi.py:851:test_runWithInteractionRaiseOriginalError
/buildslave/debian8-py2.7/Twisted/build/py27-tests/local/lib/python2.7/site-packages/twisted/enterprise/adbapi.py:341:runInteraction
/buildslave/debian8-py2.7/Twisted/build/py27-tests/local/lib/python2.7/site-packages/twisted/internet/threads.py:53:deferToThreadPool
--- <exception caught here> ---
/buildslave/debian8-py2.7/Twisted/build/py27-tests/local/lib/python2.7/site-packages/twisted/test/test_adbapi.py:725:callInThreadWithCallback
/buildslave/debian8-py2.7/Twisted/build/py27-tests/local/lib/python2.7/site-packages/twisted/enterprise/adbapi.py:477:_runInteraction
]

twisted.test.test_adbapi.ConnectionPoolTests.test_runWithInteractionRaiseOriginalError
===============================================================================
[ERROR]
Traceback (most recent call last):
  File "/buildslave/debian8-py2.7/Twisted/build/py27-tests/local/lib/python2.7/site-packages/twisted/enterprise/adbapi.py", line 304, in _runWithConnection
    conn.rollback()
  File "/buildslave/debian8-py2.7/Twisted/build/py27-tests/local/lib/python2.7/site-packages/twisted/test/test_adbapi.py", line 795, in rollback
    raise RuntimeError("problem!")
exceptions.RuntimeError: problem!

twisted.test.test_adbapi.ConnectionPoolTests.test_runWithConnectionRaiseOriginalError
===============================================================================
[ERROR]
Traceback (most recent call last):
  File "/buildslave/debian8-py2.7/Twisted/build/py27-tests/local/lib/python2.7/site-packages/twisted/enterprise/adbapi.py", line 474, in _runInteraction
    conn.rollback()
  File "/buildslave/debian8-py2.7/Twisted/build/py27-tests/local/lib/python2.7/site-packages/twisted/test/test_adbapi.py", line 838, in rollback
    raise RuntimeError("problem!")
exceptions.RuntimeError: problem!

twisted.test.test_adbapi.ConnectionPoolTests.test_runWithInteractionRaiseOriginalError
```

please see if you can update the code to have these tests pass.

thanks!

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_6"></a>|@rodrigc set owner to @adiroiban|
|-|-|

I updated the patch at:
https://github.com/twisted/twisted/pull/66

Please review

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_7"></a>|@adiroiban set owner to @rodrigc|
|-|-|

Changes look good. Thanks.

While I was reviewing the changes, I also made a few small changes, like using the string formating method used in other places in Twisted and breaking long lines without using a backslash.

Waiting for builds to be green and will merge.

Many, many thanks for the cleaup!

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_8"></a>|@adiroiban set status to `closed`|
|-|-|

In changeset 3e6382cbf5ff8312f419d81eb16c662394b3453d
```
#!CommitTicketReference repository="" revision="3e6382cbf5ff8312f419d81eb16c662394b3453d"
Merge 8345-rodrigc-pep3109: Use new syntax for raising exceptions.

Author: rodrigc
Reviewer: adiroiban
Fixes: #8345
```
